python/ray/autoscaler/_private/vsphere/pyvmomi_sdk_provider.py

```diff
@@ -67,20 +85,16 @@ def get_pyvmomi_obj_by_name(self, vimtype, name):
         if self.pyvmomi_sdk_client is None:
             raise ValueError("Must init pyvmomi_sdk_client first.")
 
-        container = self.pyvmomi_sdk_client.viewManager.CreateContainerView(
-            self.pyvmomi_sdk_client.rootFolder, vimtype, True
+        if not name:
+            raise ValueError("Invalid argument for name")
+
+        container = self.pyvmomi_sdk_client.content.viewManager.CreateContainerView(
+            self.pyvmomi_sdk_client.content.rootFolder, vimtype, True
         )
 
         for c in container.view:
-            if name:
-                if c.name == name:
-                    obj = c
-                    break
-            else:
+            if c.name == name:
                 obj = c
                 break
-        if not obj:
-            raise RuntimeError(
-                f"Unexpected: cannot find vSphere object {vimtype} with name: {name}"
-            )
```

Comment:
I noticed the same code changes needed to be made in two places; I still think it would be a bit better to unify the functions, maybe like `def get_myvmomi_obj(self, vimtype, name=None, moid=None)`, but it's not strictly related to the bug fixed in this PR and doesn't need to block the PR.

Reply:
Well, that will bring too many changes in the caller places. I suggest we make sure the behaviour are not changed for the 2 functions, and I will optimize the logic and tell team not to change them again, never. Seems changes to these 2 functions have been blockers to PRs twice.